examples/multithreading/multithreading.cpp

```diff
@@ -58,23 +61,53 @@ class VulkanExample : public VulkanExampleBase
 	};
 
 	struct ObjectData {
-		glm::mat4 model;
 		glm::vec3 pos;
-		glm::vec3 rotation;
+		float rotationAxis;
 		float rotationDir;
 		float rotationSpeed;
 		float scale;
 		float deltaT;
-		float stateT = 0;
 		bool visible = true;
+
+		struct PushConstant {
+			glm::mat4 mvp;
+			glm::vec3 color;
+		} pushConstant;
+
+		void init(vkx::Random& random) {
+			pos = random.sphere(SPHERE_SCALE);
+			rotationAxis = random.radian();
+			deltaT = random.real();
+			rotationDir = random.boolean() ? 1.0f : -1.0f;
+			// Rotate at between 2 and 6 degrees per second
+			rotationSpeed = glm::radians(random.real(2.0f, 6.0f)) * rotationDir;
+			scale = random.real(0.75f, 1.25f);
+			pushConstant.color = random.color();
+		}
+
+		void update(float frameTimer) {
+			rotationAxis += 2.5f * rotationSpeed * frameTimer;
+			rotationAxis = fmod(rotationAxis, M_TAU);
+			deltaT += 0.15f * frameTimer;
+			deltaT = fmod(deltaT, 1.0f);
+			pos.y = sin(deltaT * M_TAU) * 2.5f;
+		}
+
+		glm::mat4 getModel() {
+			auto model = glm::translate(glm::mat4(1.0f), pos);
+			model = glm::rotate(model, -sinf(deltaT * M_TAU) * 0.25f, glm::vec3(rotationDir, 0.0f, 0.0f));
```

Comment:
I've reproduced the original logic here, but I'm confused.  `glm::rotate` expects an angle here, in radiants, but we're passing the output of a `sin` function.  Two lines below we pass in `deltaT * M_TAU` without wrapping it in a `sin`.  I'm not sure what the intent is.